sdks/besu/interoperation-node-sdk/src/AssetManager.ts

```diff
@@ -0,0 +1,164 @@
+// */
+
+/**
+* This file provides helper functions for interoperability operations.
+**/
+/** End file docs */
+import * as assetLocksPb from "@hyperledger-labs/weaver-protos-js/common/asset_locks_pb";
+import  * as Web3 from 'web3-utils';
+import { Hash, SHA256 } from "./HashFunctions";
+
+// Create an asset exchange agreement structure
+function createAssetExchangeAgreementSerialized(assetType: string, assetID: string, recipientECert: string, lockerECert: string) {
+    const assetExchangeAgreement = new assetLocksPb.AssetExchangeAgreement();
+    assetExchangeAgreement.setAssettype(assetType);
+    assetExchangeAgreement.setId(assetID);
+    assetExchangeAgreement.setRecipient(recipientECert);
+    assetExchangeAgreement.setLocker(lockerECert);
+    return Buffer.from(assetExchangeAgreement.serializeBinary())
+}
+
+// Create an asset lock structure
+function createAssetLockInfoSerialized(hash: Hash, expiryTimeSecs: number)
+{
+    const lockInfoHTLC = new assetLocksPb.AssetLockHTLC();
+    lockInfoHTLC.setHashmechanism(hash.HASH_MECHANISM);
+    lockInfoHTLC.setHashbase64(Buffer.from(hash.getSerializedPreimageBase64()));
+    lockInfoHTLC.setExpirytimesecs(expiryTimeSecs);
+    lockInfoHTLC.setTimespec(assetLocksPb.TimeSpec.EPOCH)
+    return Buffer.from(lockInfoHTLC.serializeBinary())
+}
+
+// Create an asset claim structure
+function createAssetClaimInfoSerialized(hash: Hash) {
+    const claimInfoHTLC = new assetLocksPb.AssetClaimHTLC();
+    claimInfoHTLC.setHashmechanism(hash.HASH_MECHANISM);
+    claimInfoHTLC.setHashpreimagebase64(Buffer.from(hash.getSerializedPreimageBase64()));
+    const claimInfoHTLCSerialized = claimInfoHTLC.serializeBinary();
+    const claimInfo = new assetLocksPb.AssetClaim();
+    claimInfo.setLockmechanism(assetLocksPb.LockMechanism.HTLC);
+    claimInfo.setClaiminfo(claimInfoHTLCSerialized);
+    return Buffer.from(claimInfo.serializeBinary())
+}
+
+const createHTLC = async (
+    assetManagerContract: any,
+    tokenContract: any,
+    assetID: string,
+    assetData: string,
+    assetAmount: number,
+    senderAddress: string,
+    recipientAddress: string,
+    expiryTimeSecs: number,
+    hash: Hash
+): Promise< {result: any} > => {
+
+    if (!assetManagerContract) {
+        console.log("Contract handle not supplied");
+        return {result: false};
+    }
+    if (!recipientAddress) {
+        console.log(`Recipient address not supplied ${recipientAddress}`);
+        return {result: false};
+    }
+    const currTimeSecs = Math.floor(Date.now() / 1000);   // Convert epoch milliseconds to seconds
+    if (expiryTimeSecs <= currTimeSecs) {
+        console.log("Supplied expiry time invalid or in the past: %s; current time: %s", new Date(expiryTimeSecs).toISOString(), new Date(currTimeSecs).toISOString());
+        return {result: false};
+    }    
+
+    const protobufParams = createAssetExchangeAgreementSerialized("", String(assetID), recipientAddress.slice(2), Web3.utf8ToHex(String(assetData)));
```

Comment:
Shouldn't the last parameter be `senderAddress`? as that is the locker.
what is `assetData` exactly?
Also the first parameter should not be empty, have something in the asset type, take it as another function parameter and pass it here.

Also based on comment on contract, this will change and depending upon the case you should either create this, or fungible asset exchange agreement or hybrid one.